bigtable/google/cloud/bigtable/table.py

```diff
@@ -671,45 +707,83 @@ class _RetryableMutateRowsWorker(object):
     )
     # pylint: enable=unsubscriptable-object
 
-    def __init__(self, client, table_name, rows, app_profile_id=None, timeout=None):
+    def __init__(
+        self, client, table_name, rows=[], batches=[], app_profile_id=None, timeout=None
+    ):
         self.client = client
         self.table_name = table_name
         self.rows = rows
+        if rows:
+            batches = [rows]
+        self.batches = dict([(index, batch) for index, batch in enumerate(batches)])
         self.app_profile_id = app_profile_id
-        self.responses_statuses = [None] * len(self.rows)
+        self.responses_statuses = dict(
+            [(index, [None] * len(batch)) for index, batch in enumerate(batches)]
+        )
         self.timeout = timeout
+        self.batch_count = 0
+        self.num_batches = len(batches)
+        self._lock = threading.Lock()
 
     def __call__(self, retry=DEFAULT_RETRY):
-        """Attempt to mutate all rows and retry rows with transient errors.
+        """Attempt to mutate all batches and retry rows with transient errors.
 
         Will retry the rows with transient errors until all rows succeed or
         ``deadline`` specified in the `retry` is reached.
 
         :rtype: list
-        :returns: A list of response statuses (`google.rpc.status_pb2.Status`)
+        :returns: Batches of lists of response statuses (`google.rpc.status_pb2.Status`)
                   corresponding to success or failure of each row mutation
-                  sent. These will be in the same order as the ``rows``.
+                  sent. These will be in the same order as the ``rows`` in batch.
```

Comment:
The last sentence doesn't make much sense for an async operation.

Reply:
In the end thread joined, so will wait till the operation complete and retruns mapped status of rows passed in `mutate_rows()`

Reply:
And updated and removed `in batch` from sentence.